tests/client/test_client.py

```diff
@@ -165,3 +165,68 @@ def test_get_missing(client, requests_mocker):
 
     # It should explain the problem
     assert "repo1 was not found" in str(error.value)
+
+
+def test_upload_one_file(client, requests_mocker, tmpdir, caplog):
+    """test upload a file to a repo in pulp"""
+    logging.getLogger().setLevel(logging.INFO)
+    caplog.set_level(logging.INFO)
+
+    client._PULP_CHUNK_SIZE = 20
+
+    request_body = {
+        "_href": "/pulp/api/v2/content/uploads/cfb1fed0-752b-439e-aa68-fba68eababa3/",
+        "upload_id": "cfb1fed0-752b-439e-aa68-fba68eababa3",
+    }
+    upload_id = request_body["upload_id"]
+    repo_id = "repo1"
+    import_report = {
+        "result": {},
+        "error": {},
+        "spawned_tasks": [
+            {
+                "_href": "/pulp/api/v2/tasks/7744e2df-39b9-46f0-bb10-feffa2f7014b/",
+                "task_id": "7744e2df-39b9-46f0-bb10-feffa2f7014b",
+            }
+        ],
+    }
+
+    tasks_report = [
+        {"task_id": "7744e2df-39b9-46f0-bb10-feffa2f7014b", "state": "finished"}
+    ]
+
+    somefile = tmpdir.join("some-file.txt")
+    somefile.write("I am the file to be uploaded")
+
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/content/uploads/", json=request_body
+    )
+    requests_mocker.put(
+        "https://pulp.example.com/pulp/api/v2/contents/%s/0" % upload_id, json=[]
+    )
+    requests_mocker.put(
+        "https://pulp.example.com/pulp/api/v2/contents/%s/20" % upload_id, json=[]
+    )
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/repositories/%s/actions/import_upload/"
+        % repo_id,
+        json=import_report,
+    )
+    requests_mocker.delete(
+        "https://pulp.example.com/pulp/api/v2/content/uploads/%s/" % upload_id, json=[]
+    )
+    requests_mocker.post(
+        "https://pulp.example.com/pulp/api/v2/tasks/search/", json=tasks_report
+    )
+
+    client.upload_one_file(str(somefile), repo_id, "txt", {})
+
+    messages = caplog.messages
+    lines = [m.splitlines()[0] for m in messages]
+
+    # 6 reuqests should be made
+    assert requests_mocker.call_count == 6
+
+    # task's spwaned and completed
+    assert "Created Pulp task: 7744e2df-39b9-46f0-bb10-feffa2f7014b" in messages
+    assert "Pulp task completed: 7744e2df-39b9-46f0-bb10-feffa2f7014b" in messages
```

Comment:
I think there's some verification missing here, like verifying the specifics of what was passed into import_upload and verifying that the upload was deleted.

Reply:
got it.